### ISSUE TITLE
Type definitions missing in plain JavaScript

### ISSUE THREAD

**Issue by tkesgar** [Microsoft/vscode#30908]
- VSCode Version: 1.14.1
- OS Version: Windows 10 Pro

After updating Visual Studio Code to 1.14.1, I can't seem to use type definitions on JavaScript projects.

In this example below, previously I can simply use `Knex` class without further configuration after installing the package through `npm`.

![image](https://user-images.githubusercontent.com/2698543/28304780-5da53cbc-6bc3-11e7-943b-612dc31618e3.png)

Also, it seems that type detections from packages are missing. Previously VSCode can detect the return type from functions in npm packages; now it can't.

![image](https://user-images.githubusercontent.com/2698543/28305234-1626d7ae-6bc5-11e7-8d32-df71f5fc0ac3.png)

However, functions returning standard JavaScript types are still shown properly.

![image](https://user-images.githubusercontent.com/2698543/28305538-2d0692a6-6bc6-11e7-8545-ff6eb8e0d358.png)

My project is a "plain standard" npm project (no TypeScript), with `package.json` present and all modules installed in `node_modules`.

Is this a bug or there are new changes? If there are new changes, is there any configurations or plugins required to get the type definitions back?

Steps to Reproduce:
1. Update to 1.14.1
2. Open a JavaScript project directory containing a npm package.
3. Attempt to use the type in IntelliSense, or look at definitions from a package function.

Reproduces without extensions: Yes

**Comment by mjbvz** [comment]
To help investigate, please:

1. Open the output panel and go the `TypeScript` section. Do you see any error messages printed in there?
2. Run the`Toggle Developer Tools` command and open the developer console. Do you see any errors printed in there?

Thanks

**Comment by tkesgar** [comment]
Hello,

I opened the `TypeScript` section and saw this:
```
[Info  - 09.52.06] Using tsserver from: c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js
[Info  - 09.52.06] Forking TSServer
PATH: undefined
```

As for the `Toggle Developer Tools`, there are a number of warnings about "confused snippet" but no errors.

![image](https://user-images.githubusercontent.com/2698543/28398557-e646c7ec-6d31-11e7-826f-200a05f016be.png)

**Comment by tkesgar** [comment]
I managed to get the typings back after removing `node_modules` folder in the project and performing `npm install`.

This is after I tried to reproduce the issue with a blank new project folder and installing some modules, but found that the typings are present. It seems that the issue is related to my project folder.

Closing this issue for now; sorry for spending your time, and thank you for maintaining VS Code.
